src/proj.rs

```diff
@@ -557,12 +569,36 @@ impl Proj {
             } else {
                 None
             };
-            let name = if !name.is_null() {
-                Some(_string(name)?)
+            Ok((area, name))
+        }
+    }
+
+    fn pj_info(&self) -> PjInfo {
+        unsafe {
+            let pj_info = proj_pj_info(self.c_proj);
+            let id = if pj_info.id.is_null() {
+                None
             } else {
+                Some(_string(pj_info.id).expect("PROJ built an invalid string"))
+            };
```

Comment:
With the 1.50.0 release, we could use [`bool::then`](https://doc.rust-lang.org/stable/std/primitive.bool.html#method.then) here, but I think it might be a little too soon to avoid breakage for downstream users. We should codify our policy around Rust versions at some point...

Reply:
Oh right. rust versions. 😄

Reply:
Too soon! But yes, we should.